static/js/message_events.js

```diff
@@ -391,12 +422,9 @@ exports.update_messages = function update_messages(events) {
 };
 
 exports.remove_messages = function (message_ids) {
-    for (const list of [message_list.all, home_msg_list, message_list.narrowed]) {
-        if (list === undefined) {
-            continue;
-        }
-        list.remove_and_rerender(message_ids);
-    }
+    stored_mld.forEach((mld) => mld.remove(message_ids));
+    all_message_lists().forEach((ml) => ml.remove_and_rerender(message_ids));
```

Comment:
I don't understand the logic here -- we're doing a removal on all message lists first, and then doing remove-and-rerender afterwards?  I think that'll be fragile or broken if a message list is in both sets, as the check for whether rerendering is required might be missed.

Would it work better to have these operations in the opposite order?

Reply:
I reversed the order for consistency with the other changes.
But the above order wouldn't be a problem since `remove_and_rerender` does a force `render`.